pkg/operator/controller/dns_status_test.go

```diff
@@ -212,140 +223,184 @@ func TestComputeDNSDegradedCondition(t *testing.T) {
+			name:          "too few DNS pods available (integer)",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(6, 4, intstr.FromInt(1)),
+			oldCondition:  defaultOldCondition,
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 5,
+			expectRequeue: true,
 		},
 		{
 			name:         "enough available (percentage)",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(100, 90, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(100, 67, intstr.FromString("33%")),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
 		{
 			name:         "enough available (integer)",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(6, 5, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 5, intstr.FromInt(1)),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
 		{
 			name:         "all available",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromString("33%")),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
+		{
+			name:          "not enough DNS pods available, check requeue",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(10, 8, intstr.FromString("10%")),
+			oldCondition:  defaultOldCondition,
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 5,
+			expectRequeue: true,
+		},
+		{
+			name:          "no DNS pods available for <5m",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(6, 0, intstr.FromString("10%")),
+			oldCondition:  cond(DNSControllerDegradedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-4)),
+			expected:      operatorv1.ConditionTrue,
+			expectRequeue: false,
+		},
+		{
+			name:          "not enough DNS pods available for <5m",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(10, 8, intstr.FromString("10%")),
+			oldCondition:  cond(DNSControllerDegradedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-4)),
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 1,
+			expectRequeue: true,
+		},
+		{
+			name:          "no DNS pods available for >5m",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(6, 0, intstr.FromString("10%")),
+			oldCondition:  cond(DNSControllerDegradedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-6)),
+			expected:      operatorv1.ConditionTrue,
+			expectRequeue: false,
+		},
+		{
+			name:          "not enough DNS pods available for >5m",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(10, 8, intstr.FromString("10%")),
+			oldCondition:  cond(DNSControllerDegradedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-6)),
+			expected:      operatorv1.ConditionFalse,
```

Comment:
I thought the point of the original issue reported in Bugzilla and the point of this PR was that in the "not enough DNS pods available for <5m" case, the operator should report `Degraded=False`, and in the "not enough DNS pods available for >5m" case, the operator should report `Degraded=True`.  Why does this test case expect `Degraded=False`?
```suggestion
			expected:      operatorv1.ConditionTrue,
```

Reply:
Whenever it is going to requeue, it will set Degraded=False and set the requeue wait to at most 5 minutes.  It will always requeue if not enough DNS pods are available, no matter how long they've been unavailable.  All I'm checking in this test is that the requeue wait time is adjusted.  The point is to requeue and try to  recalculate the degraded condition while there aren't enough pods, reporting degraded=false in the meantime.  In the case that there are NO pods available, it's always degraded=true.

Reply:
In a scenario in which some pods are available but not enough pods are available, the operator should allow for the grace period and then set `Degraded=True` when the grace period expires.  This test case looks like exactly that scenario and condition (not enough pods, grace period has expired).  Expecting a requeue with the appropriate interval is fine, but why do we expect `Degraded=False`?  Do we have a test case that verifies that `computeDNSDegradedCondition` sets `Degraded=True` when the grace period expires?

Reply:
Fixed, it now sets `Degraded=True` when not enough pods are available for > 5 min